File: CapstoneProj/Services/PurchaseService.cs
using CapstoneProj.Repository.Interface;
using CapstoneProj.Services.Interface;
using CapstoneProj.Services;
using CapstoneProj.Models;
namespace CapstoneProj.Services
{
    public class PurchaseService : IPurchaseService
    {
        private readonly IPurchaseRepo _purchaseRepo;

        public PurchaseService(IPurchaseRepo purchaseRepo)
        {
            _purchaseRepo = purchaseRepo;
        }

        public IEnumerable<Purchase> GetAllPurchases()
        {
            return _purchaseRepo.GetAllPurchases();
        }

        public Purchase GetPurchaseById(int purchaseId)
        {
            return _purchaseRepo.GetPurchaseById(purchaseId);
        }

        public Purchase AddPurchase(Purchase purchase)
        {
            return _purchaseRepo.AddPurchase(purchase);
        }

        public Purchase UpdatePurchase(Purchase purchase)
        {
            return _purchaseRepo.UpdatePurchase(purchase);
        }

        public Purchase DeletePurchase(int purchaseId)
        {
            return _purchaseRepo.DeletePurchase(purchaseId);
        }
    }
}

File: src/app/Services/purchase.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Purchase } from '../Models/PurchaseModel';

@Injectable({
  providedIn: 'root'
})
export class PurchaseService {
  private apiUrl = 'https://localhost:7094/api/Purchase'; 

  constructor(private http: HttpClient) {}

  getAllPurchases(): Observable<Purchase[]> {
    return this.http.get<Purchase[]>(this.apiUrl);
  }

  getPurchaseById(purchaseId: number): Observable<Purchase> {
    return this.http.get<Purchase>(`${this.apiUrl}/${purchaseId}`);
  }

  addPurchase(purchase: Purchase): Observable<Purchase> {
    return this.http.post<Purchase>(this.apiUrl, purchase);
  }

  updatePurchase(purchase: Purchase): Observable<Purchase> {
    return this.http.put<Purchase>(`${this.apiUrl}/${purchase.purchaseId}`, purchase);
  }

  deletePurchase(purchaseId: number): Observable<any> {
    return this.http.delete(`${this.apiUrl}/${purchaseId}`);
  }
}